aws_lambda_builders/workflows/nodejs_npm_esbuild/actions.py

```diff
@@ -62,138 +55,35 @@ def __init__(
         self.osutils = osutils
         self.subprocess_esbuild = subprocess_esbuild
         self.skip_deps = skip_deps
-        self.subprocess_nodejs = subprocess_nodejs
+        self.manifest = manifest
 
     def execute(self):
         """
         Runs the action.
 
         :raises lambda_builders.actions.ActionFailedError: when esbuild packaging fails
         """
+        esbuild_command_builder = EsbuildCommandBuilder(
+            self.scratch_dir, self.artifacts_dir, self.bundler_config, self.osutils, self.manifest
+        )
+        esbuild_command_builder.build_entry_points()
+        esbuild_command_builder.build_default_values()
 
-        if self.ENTRY_POINTS not in self.bundler_config:
-            raise ActionFailedError(f"{self.ENTRY_POINTS} not set ({self.bundler_config})")
-
-        entry_points = self.bundler_config[self.ENTRY_POINTS]
-
-        if not isinstance(entry_points, list):
-            raise ActionFailedError(f"{self.ENTRY_POINTS} must be a list ({self.bundler_config})")
-
-        if not entry_points:
-            raise ActionFailedError(f"{self.ENTRY_POINTS} must not be empty ({self.bundler_config})")
-
-        entry_paths = [self.osutils.joinpath(self.scratch_dir, entry_point) for entry_point in entry_points]
-
-        LOG.debug("NODEJS building %s using esbuild to %s", entry_paths, self.artifacts_dir)
-
-        explicit_entry_points = []
-        for entry_path, entry_point in zip(entry_paths, entry_points):
-            explicit_entry_points.append(self._get_explicit_file_type(entry_point, entry_path))
+        # Handle the event of marking all dependencies as external
+        if self.skip_deps or "./node_modules/*" in self.bundler_config.get("external", []):
```

Comment:
Does this cover cases where node_modules may be hoisted further up (such as in a yarn workspaces repository) e.g. `../../node_modules/*`?

Reply:
Unfortunately it wouldn't, but right now lambda builders only installs the dependencies it finds in the manifest at the CodeURI root, so I don't imagine this would break anything. I image there's some area for improvement here though.